controller/volume.go

```diff
@@ -45,6 +45,9 @@ type Qualifier interface {
 	// ShouldProvision returns whether provisioning for the claim should
 	// be attempted.
 	ShouldProvision(*v1.PersistentVolumeClaim) bool
+
+	// ShouldDelete returns whether deleting the PV should be attempted.
+	ShouldDelete(volume *v1.PersistentVolume) bool
```

Comment:
can we make a new interface instead? just to avoid breaking current implementers so they aren't forced to write a ShoulDelete, also to allow people to implement only one or the other. I am not sure what to call it though...

Reply:
good point. 
I thought for a while about the name.. and think to call it DeletionGuard.
let me know if it sounds good.

Reply:
yeah, sounds good to me :+1: (anyway the name can maybe be changed someday...just the method can't)